bigtable/client/version.h

```diff
@@ -15,9 +15,11 @@
 #ifndef GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_VERSION_H_
 #define GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_VERSION_H_
 
+#ifdef __linux
```

Comment:
That won't work when ported to macOS, or FreeBSD, or any other Unix out there.  Please consider:

```C++
#ifndef _MSC_VER
// Microsoft Visual C++ does not define the __cplusplus macro correctly.
// We only use it on other compilers:
#if __cplusplus < 201103L
#error "Bigtable C++ Client requires C++11"
#endif  // __cplusplus < 201103L
#else
// Check the MSVC version explicitly, based on the data from:
//   http://en.cppreference.com/w/cpp/compiler_support
#if _MSC_VER < 1900 /* TODO() figure this out, I think 1700 is safe */
#error "Bigtable C++ Client requires C++11 features not supported this version of MSVC"
#endif  // _MSC_VER
```

Check with http://en.cppreference.com/w/cpp/compiler_support to find out what compiler we need.